Formula/shogun.rb

```diff
@@ -0,0 +1,115 @@
+  end
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "swig" => :build
+  depends_on "arpack"
+  depends_on "eigen"
+  depends_on "glpk"
+  depends_on "hdf5"
+  depends_on :java => ["1.7+", :build]
+  depends_on "json-c"
+  depends_on "lapack"
+  depends_on "lzo"
+  depends_on "nlopt"
+  depends_on "python" if MacOS.version <= :lion
+  depends_on "readline"
+  depends_on "snappy"
+  depends_on "xz"
+
+  resource "jblas" do
+    url "https://mikiobraun.github.io/jblas/jars/jblas-1.2.3.jar"
+    sha256 "e9328d4e96db6b839abf50d72f63626b2309f207f35d0858724a6635742b8398"
+  end
+
+  resource "numpy" do
+    url "https://pypi.python.org/packages/ee/66/7c2690141c520db08b6a6f852fa768f421b0b50683b7bbcd88ef51f33170/numpy-1.14.0.zip"
+    sha256 "3de643935b212307b420248018323a44ec51987a336d1d747c1322afc3c099fb"
+  end
+
+  needs :cxx11
+
+  def install
+    ENV.cxx11
+    # fix build of modular interfaces with SWIG 3.0.5 on MacOSX
+    # https://github.com/shogun-toolbox/shogun/pull/2694
+    # https://github.com/shogun-toolbox/shogun/commit/fef8937d215db7
+    ENV.append_to_cflags "-D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0"
+
+    ENV["PYTHONPATH"] = libexec/"lib/python2.7/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+
+    resource("numpy").stage do
+      system "python", *Language::Python.setup_install_args(libexec/"vendor")
+    end
+
+    system "python", *Language::Python.setup_install_args(libexec)
+    bin.install Dir[libexec/"bin/*"]
+    env = {
+      :PATH => "#{libexec}/vendor/bin:$PATH",
+      :PYTHONPATH => ENV["PYTHONPATH"],
+      :LAPACKE_PATH => '#{Formula["lapack"].opt_lib}',
+    }
+    bin.env_script_all_files(libexec/"bin", env)
+
+    libexec.install resource("jblas")
+
+    args = std_cmake_args + [
+      "-DBUILD_EXAMPLES=OFF",
+      "-DBUNDLE_JSON=OFF",
+      "-DBUNDLE_NLOPT=OFF",
+      "-DENABLE_TESTING=OFF",
+      "-DENABLE_COVERAGE=OFF",
+      "-DBUILD_META_EXAMPLES=OFF",
+      "-DINTERFACE_PYTHON=ON",
+      "-DINTERFACE_JAVA=ON",
+      "-DJBLAS=#{libexec}/jblas-#{resource("jblas").version}.jar",
+      "-DLIB_INSTALL_DIR=#{lib}",
+    ]
+
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make", "install"
+    end
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+        #include <cstdlib>
+        #include <cstring>
+        #include <assert.h>
+
+        #include <shogun/base/init.h>
+        #include <shogun/lib/versionstring.h>
+
+        using namespace shogun;
+
+        int main(int argc, char** argv)
+        {
+          init_shogun_with_defaults();
+          assert (std::strcmp(MAINVERSION, "#{version}") == 0);
+          exit_shogun();
+
+          return EXIT_SUCCESS;
+        }
+      EOS
+
+    system ENV.cxx, "-std=c++11", "-I#{include}", "test.cpp", "-o", "test", "-L#{lib}", "-lshogun"
+    system "./test"
+  end
+end
+
```

Comment:
looks like an extra line

Reply:
you guys dont need the extra line on the end of the file, i.e. [POSIX standard](https://stackoverflow.com/a/729795/2798875)? 😕

Reply:
Ah it's just the GitHub GUI playing tricks. I thought you had an extra blank line at the end.